src/com/seafile/seadroid2/cameraupload/CameraUploadService.java

```diff
@@ -228,33 +230,32 @@ private File getPhotoFromMediaStore(Context context, Uri uri) {
                 e.printStackTrace();
             }
 
-            return CameraUploadUtil.getAllPhotosAbsolutePathList();
-        }
+            List<File> pathList = CameraUploadUtil.getAllPhotosAbsolutePathList();
 
-        @Override
-        protected void onPostExecute(List<File> result) {
-            Intent localIntent;
-            if (result == null) {
-                if (!isNetworkAvailable) {
-                    // do nothing until network connection available
-                }
-                if (isNetworkAvailable && !isRemoteCameraUploadRepoValid) {
-                    localIntent = new Intent(TransferManager.BROADCAST_ACTION).putExtra("type",
-                            BROADCAST_CAMERA_UPLOAD_LIBRARY_NOT_FOUND);
-                    LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(localIntent);
+            if (pathList != null) {
+                for (File photo : pathList) {
+                    String path = photo.getName();
+                    // use local database to detect duplicate upload
+                    SeafCachedPhoto cp = cUploadManager.getCachedPhoto(repoName, repoId, DIR, path);
+                    if (cp == null) {
+                        // add photos to uploading queue
+                        addCameraUploadTask(repoId, repoName, CAMERA_UPLOAD_REMOTE_PARENTDIR + CAMERA_UPLOAD_REMOTE_DIR, photo.getAbsolutePath());
```

Comment:
So now we're adding the tasks to the queue in another thread. Is this safe to concurrently add tasks from two threads? maybe we need to add `synchronzied` to serialize access to `TranferManager.addTaskToQue`?